x-pack/plugins/security_solution/public/detection_engine/rule_creation/components/related_integrations/related_integrations.test.tsx

```diff
@@ -0,0 +1,698 @@
+    });
+
+    it.each([
+      [
+        'Not installed',
+        {
+          package_name: 'package-a',
+          package_title: 'Package A',
+          latest_package_version: '1.2.0',
+          is_installed: false,
+          is_enabled: false,
+        },
+      ],
+      [
+        'Installed: Disabled',
+        {
+          package_name: 'package-a',
+          package_title: 'Package A',
+          installed_package_version: '1.2.0',
+          latest_package_version: '1.2.0',
+          is_installed: true,
+          is_enabled: false,
+        },
+      ],
+      [
+        'Installed: Enabled',
+        {
+          package_name: 'package-a',
+          package_title: 'Package A',
+          installed_package_version: '1.2.0',
+          latest_package_version: '1.2.0',
+          is_installed: true,
+          is_enabled: true,
+        },
+      ],
+    ])('shows integration status "%s" in combo box popover', async (status, integrationData) => {
+      fleetIntegrationsApi.fetchAllIntegrations.mockResolvedValue({
+        integrations: [integrationData],
+      });
+
+      render(<TestForm />, { wrapper: createReactQueryWrapper() });
+
+      await addRelatedIntegrationRow();
+      await showEuiComboBoxOptions(screen.getByTestId(COMBO_BOX_TOGGLE_BUTTON_TEST_ID));
+
+      expect(screen.getByRole('option')).toHaveTextContent(status);
+    });
+
+    it.each([
+      [
+        'Package A',
+        {
+          package_name: 'package-a',
+          package_title: 'Package A',
+          latest_package_version: '1.2.0',
+          is_installed: false,
+          is_enabled: false,
+        },
+      ],
+      [
+        'Package A: Disabled',
+        {
+          package_name: 'package-a',
+          package_title: 'Package A',
+          installed_package_version: '1.2.0',
+          latest_package_version: '1.2.0',
+          is_installed: true,
+          is_enabled: false,
+        },
+      ],
+      [
+        'Package A: Enabled',
+        {
+          package_name: 'package-a',
+          package_title: 'Package A',
+          installed_package_version: '1.2.0',
+          latest_package_version: '1.2.0',
+          is_installed: true,
+          is_enabled: true,
+        },
+      ],
+    ])(
+      'shows integration name with its status "%s" when selected in combo box',
+      async (nameWithStatus, integrationData) => {
+        fleetIntegrationsApi.fetchAllIntegrations.mockResolvedValue({
+          integrations: [integrationData],
+        });
+
+        render(<TestForm />, { wrapper: createReactQueryWrapper() });
+
+        await addRelatedIntegrationRow();
+        await selectFirstEuiComboBoxOption({
+          comboBoxToggleButton: screen.getByTestId(COMBO_BOX_TOGGLE_BUTTON_TEST_ID),
+        });
+
+        expect(screen.getByTestId(COMBO_BOX_SELECTION_TEST_ID)).toHaveTextContent(
+          new RegExp(`^${nameWithStatus}$`)
```

Comment:
I'm wondering why it doesn't match without using RegExp?

Reply:
This is [how toHaveTextContent() works](https://github.com/testing-library/jest-dom#tohavetextcontent).